recipes/imds_config.rb

```diff
@@ -49,3 +49,7 @@
     raise "head_node_imds_secured must be 'true' or 'false', but got #{node['cluster']['head_node_imds_secured']}"
   end
 end
+
+# IMDS access must be verified on cluster creation, but skipped on cluster update
+# because customer may have introduced IMDS access customizations.
```

Comment:
even if the customer introduced IMDS access customizations, the fact remains that the cluster is not configured as the parameter dictates it should be. for that reason i think we should continue to fail after update.

there are also two other alternatives that i can think:
* provide some type of warning message to the user that performed the update. (this seems hard because it would seem to require somehow passing the message through CFN)
* don't support update of the IMDS restriction parameter

Reply:
I think the best solution would be to return a warning message because:
1. We should support the update of something as long as it is technically feasible, as it is a useful feature for customers
2. It is informative without causing frustration to the customer: a hard fail could make the customer be stuck and it can be frustrating

A hard fail would be right in case the customer can have full fine-grained control over IMDS lockdown though the config file. Le's think about the integration with OpsWorks: in that case the customer must manually allow-list the aws user to access IMDS, so a failing update would be really frustrating because every time she/he should remove and re-add the user to make update succeed.

Reply:
Talked with @fdm about this offline. He proposed to keep it simple and remove the support to the update. I sounded to me a bit extreme solution, but it is true that it would simplify a lot. We are going to introduce it later on if customer need it.